### PR TITLE
Fix for integration shell tests

### PR DESCRIPTION
**Pull request by xeacott**
test_shell_disabled and test_shell_default_enabled are now behaving
properly with updated test command.

### What does this PR do?
Uses echo instead of printf based on docs for wc l flag

### What issues does this PR fix or reference?
#52899 
#52898 

### Previous Behavior
test_shell_disabled and test_shell_default_enabled would fail as wc had different behavior when using
the l flag

### New Behavior
now these tests pass using echo and newline character

### Tests written?
**[NOTICE] Bug fixes or features added to Salt require tests.**
Please review the [test documentation](https://docs.saltstack.com/en/latest/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite.

Yes

### Commits signed with GPG?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: tests/integration/shell/test_enabled.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ class EnabledTest(ModuleCase):
     validate the use of shell processing for cmd.run on the salt command line
     and in templating
     '''
-    cmd = ("printf '%s\n' first second third | wc -l ; "
+    cmd = ("echo -e 'first\\nsecond\\nthird' | wc -l ; "
```
Comment (twangboy) [inline tests/integration/shell/test_enabled.py]: Should it be `\\n` as you have it here or `\n`?

Comment (xeacott) [reply tests/integration/shell/test_enabled.py]: It should remain as is since echo is being used instead of printf and they handle EOL differently

**Review by cmcmarrow — APPROVED** [review]

**Comment by xeacott** [comment]
re-run docs

**Comment by xeacott** [comment]
re-run full all

**Review by Akm0d — APPROVED** [review]

**Comment by sagetherage** [comment]
@xeacott is this still valid to merge into develop or should we be putting this into master or maybe we already did?

**Pull request closed without merging**